models/errors_whitebox_test.go

```diff
@@ -0,0 +1,30 @@
+package models
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/almighty/almighty-core/resource"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestSimpleError_Error(t *testing.T) {
+	t.Parallel()
+	resource.Require(t, resource.UnitTest)
+	e := simpleError{message: "foo"}
+	assert.Equal(t, "foo", e.Error())
+}
+
+func TestBadParameterError_Error(t *testing.T) {
+	t.Parallel()
+	resource.Require(t, resource.UnitTest)
+	e := BadParameterError{parameter: "foo", value: "bar"}
+	assert.Equal(t, fmt.Sprintf(stBadParameterErrorMsg, e.parameter, e.value), e.Error())
+}
+
+func TestNotFoundError_Error(t *testing.T) {
+	t.Parallel()
+	resource.Require(t, resource.UnitTest)
+	e := NotFoundError{entity: "foo", ID: "bar"}
+	assert.Equal(t, fmt.Sprintf(stNotFoundErrorMsg, e.entity, e.ID), e.Error())
```

Comment:
I test the format, yes. But I'm also testing the errors as a unit. And even if we don't have a strict format right now. We might have it in the future and then we have a test for this. I strongly believe in what you said about scenarios and paths. But then I'm doing an integration test. So far I just wanted to test the errors unit. We can still layer tests that go through different packages as we go.